backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/FunctionReferenceLowering.kt

```diff
@@ -240,12 +301,32 @@ internal class CallableReferenceLowering(val context: Context): FileLoweringPass
             }
 
             val constructor = buildConstructor()
-            if (!isKSuspendFunction)
+            val functionInvoke = if (isKSuspendFunction)
+                null
+            else
                 buildInvokeMethod(functionClass.getInvokeFunction())
-            suspendFunctionClass?.let {
-                val invokeMethod = buildInvokeMethod(it.getInvokeFunction())
-                if (isKSuspendFunction)
-                    invokeMethod.overriddenSymbols += functionClass.getInvokeFunction().symbol
+            val suspendFunctionInvoke = if (suspendFunctionClass == null)
+                null
+            else {
+                buildInvokeMethod(suspendFunctionClass.getInvokeFunction()).also {
+                    if (isKSuspendFunction)
+                        it.overriddenSymbols += functionClass.getInvokeFunction().symbol
+                }
+            }
+            samSuperType?.let { superTypes += it }
+            val sam = samSuperClass?.functions?.single { it.owner.modality == Modality.ABSTRACT }
+            if (sam != null) {
+                if (sam.owner.extensionReceiverParameter != null)
+                    buildInvokeMethod(sam.owner)
```

Comment:
Why can't we reuse the existing invoke in this case as below?

Reply:
We can, but some work needs to be done in bridges building algorithm - it doesn't expect extension receiver to hop into value parameters.

Reply:
I see. Ok, let's keep it this way for now.